addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1064 @@
+        select = map(lambda x: (x, calendar_id2real_id(x)), self.ids)
+        result = []
+        if self.ids:
+            self.invalidate_cache()
+        real_data = super(CalendarEvent, self.browse([real_id for calendar_id, real_id in select])).read(fields=fields2, load=load)
+        real_data = dict(zip([x['id'] for x in real_data], real_data))
+
+        for calendar_id, real_id in select:
+            res = real_data[real_id].copy()
+            ls = calendar_id2real_id(calendar_id, with_date=res and res.get('duration', 0) > 0 and res.get('duration') or 1)
+            if not isinstance(ls, (basestring, int, long)) and len(ls) >= 2:
+                res['start'] = ls[1]
+                res['stop'] = ls[2]
+
+                if res['allday']:
+                    res['start_date'] = ls[1]
+                    res['stop_date'] = ls[2]
+                else:
+                    res['start_datetime'] = ls[1]
+                    res['stop_datetime'] = ls[2]
+
+                if 'display_time' in fields:
+                    res['display_time'] = self._get_display_time(ls[1], ls[2], res['duration'], res['allday'])
+
+            res['id'] = calendar_id
+            result.append(res)
+
+        for r in result:
+            if r['user_id']:
+                user_id = type(r['user_id']) in (tuple, list) and r['user_id'][0] or r['user_id']
+                if user_id == self.env.uid:
+                    continue
+            if r['privacy'] == 'private':
+                for f in r.keys():
+                    if f not in ('id', 'allday', 'start', 'stop', 'duration', 'user_id', 'state', 'interval', 'count', 'recurrent_id_date', 'rrule'):
+                        if isinstance(r[f], list):
+                            r[f] = []
+                        else:
+                            r[f] = False
+                    if f == 'name':
+                        r[f] = _('Busy')
+
+        for r in result:
+            for k in EXTRAFIELDS:
+                if (k in r) and (fields and (k not in fields)):
+                    del r[k]
+        if isinstance(self.ids, (basestring, int, long)):
+            return result and result[0] or False
+        return result
+
+    @api.multi
+    def unlink(self, can_be_deleted=True):
+        res = False
+
+        ids_to_exclure = []
+        ids_to_unlink = []
+
+        for event_id in self.ids:
+            if len(str(event_id).split('-')) == 1:  # if  ID REAL
+                if self.browse(int(event_id)).recurrent_id:
+                    ids_to_exclure.append(event_id)
+                else:
+                    ids_to_unlink.append(int(event_id))
+            else:
+                ids_to_exclure.append(event_id)
+
+        if ids_to_unlink:
+            res = super(CalendarEvent, self.browse(ids_to_unlink)).unlink()
+
+        if ids_to_exclure:
+            for id_to_exclure in ids_to_exclure:
+                res = self.browse(id_to_exclure).write({'active': False})
```

Comment:
If you want to write same value on all ids_to_exclure records then what is need of loop here ?